lib/core/macros.nim

```diff
@@ -1344,6 +1348,93 @@ proc copy*(node: NimNode): NimNode {.compileTime.} =
   ## An alias for copyNimTree().
   return node.copyNimTree()
 
+type GenAstOpt* = enum
+  kNoExposeLocalInjects,
+    # when unset, inject'd symbols (including implicit ones such as local procs
+    # in scope) are exposed implicitly;
+    # gensym'd symbols will generate a CT internal error: `environment misses:`
+    # when set, local symbols are not exposed unless captured explicitly in
+    # `genAst` argument list. The default is unset, to avoid surprising hijacking
+    # of local symbols by symbols in caller scope.
```

Comment:
This comment should be rephrased. It is really hard for me to read and understand it.

Reply:
improved it in latest commit